webpush-subscription-2022.bs

```diff
@@ -0,0 +1,360 @@
+</div>
+
+# Introduction # {#introduction}
+
+*This section is non-normative.*
+
+The [[!SOLID-NOTIFICATIONS inline]] describes a general pattern by which agents can be notified when a Solid Resource changes.
+
+This document describes a Solid Notifications subscription type that makes use of the [[!PUSH-API inline]] for Web Push notifications in Progressive Web Applications (PWAs).
+
+This specification is for:
+
+* Resource server developers who wish to enable clients, i.e., PWAs, to listen for updates to particular resources.
+* Application developers who wish to implement a client, i.e., a PWA, to listen for updates to particular resources.
+
+## Terminology ## {#terminology}
+
+*This section is non-normative.*
+
+This document uses terminology from the [[!SOLID-NOTIFICATIONS]] protocol, 
+  including "topic".
+
+Issue(62):
+
+Note: Let the predicate `topic` be an `rdfs:subClassOf` of `as:object`?
+
+
+
+The document uses terms from [[!PUSH-API]] specification, 
+  including "push endpoint", "push service" and "authentication secret".
+It also uses terms from [[!OAUTH-2.0]] specification,
+  including  "authorization server" and "access token".
+In addition, the document uses terms from the [[!WEBARCH]] specification, 
+  including "information resource".
+
+This document uses the following terms as defined below:
+: subscription service
+:: A `subscription service` is a service, which handles and manages subscriptions for resources stored on the corresponding Solid Storage. 
+:: The `subscription service` is identified by a URI. Dereferencing this URI yields the `subscription service`'s representation, a discription of the service. By this representation, information may be provided such as subscription endpoints where subscriptions can be submitted for the service to process.
+
+Issue(63):
+
+Issue(36):
+
+: browser messaging service
+:: Refers to the implementation of a "push service" [[!PUSH-API]] in a browser.
+
+
+## Overview ## {#overview}
+
+The following diagram shows the high-level interactions involved in this flow. 
+How a client retrieves an access token is outside the scope of this document.
+
+<figure>
+  <img src="webpush-subscription-2022-flow.mmd.svg" rel="schema:image" width="800" />
+  <figcaption property="schema:name">Solid WebPushSubscription2022 Flow</figcaption>
+</figure>
+
+<ul>
+<li> 
+**Discovery:**
+The *subscriber* discovers from the *storage metadata* a suitable *subscription service*.
+It further discovers from the *subscription service*'s representation a suitable subscription endpoint to submit subscription requests to and the `vapidPublicKey` of the *subscription service*.
+</li>
+<li> 
+**Establish Subscription:**
+The *subscriber* subscribes to the *browser messaging service* to receive Web Push notifications using the `vapidPublicKey` of the *subscription service*.
+In return, the *subscriber* retrieves the `endpoint`, `auth` and `p256dh` values. 
```

Comment:
We're trying to move away from the use of `endpoint`. Depending on the context, we generally use `subscription`, `topic` or `source`. It seems that `subscription` might be best suited here, but I'll leave that up to you.

Reply:
This particular `endpoint` is the "Push Endpoint" of the PUSH API. Discussion in the other comment on code.

However, I did use the term `subscription endpoint` which I now have replaced with `Notification Subscription API` as used by the Notifications Protocol.